sdk/mobile-center-crashes/src/main/java/com/microsoft/azure/mobile/crashes/ErrorAttachments.java

```diff
@@ -22,52 +25,75 @@
+    public static ErrorAttachmentLog attachmentWithText(String text, String fileName) {
+        return attachment(null, text, fileName, CONTENT_TYPE_TEXT_PLAIN);
     }
 
     /**
-     * Build an attachment with binary suitable for using in {link CrashesListener#getErrorAttachment(ErrorReport)}.
+     * Build an error attachment log with binary suitable for using in {link CrashesListener#getErrorAttachments(ErrorReport)}.
      *
-     * @param data        binary data.
-     * @param fileName    file name to use on reports.
-     * @param contentType data MIME type.
-     * @return error attachment or null if null data is passed.
+     * @param data     binary data.
+     * @param fileName file name to use in error attachment log.
+     * @return ErrorAttachmentLog attachment or null if null data is passed.
      */
-    public static ErrorAttachment attachmentWithBinary(byte[] data, String fileName, String contentType) {
-        return attachment(null, data, fileName, contentType);
+    public static ErrorAttachmentLog attachmentWithBinary(byte[] data, String fileName, String contentType) {
+        return attachment(data, null, fileName, contentType);
     }
 
     /**
-     * Build an attachment with text and binary suitable for using in {link CrashesListener#getErrorAttachment(ErrorReport)}.
+     * Build an error attachment log with text and binary suitable for using in {link CrashesListener#getErrorAttachments(ErrorReport)}.
      *
-     * @param text        text data.
      * @param data        binary data.
-     * @param fileName    file name to use on reports for the binary data.
+     * @param text        text to attach to the attachment log.
+     * @param fileName    file name to use in error attachment log.
      * @param contentType binary data MIME type.
-     * @return error attachment or null if text and data are null.
+     * @return ErrorAttachmentLog attachment or null if text and data are null.
+     */
+    private static ErrorAttachmentLog attachment(byte[] data, String text, String fileName, String contentType) {
+        String content = text;
+        if (isBinaryContentType(contentType)) {
+            content = Base64.encodeToString(data, Base64.DEFAULT);
+        }
+        ErrorAttachmentLog attachmentLog = new ErrorAttachmentLog();
+        attachmentLog.setContentType(contentType);
+        attachmentLog.setFileName(fileName);
+        attachmentLog.setData(content);
+        return attachmentLog;
+    }
+
+    /**
+     * Checks if content type provided by user is binary.
+     *
+     * @param contentType content type.
+     * @return True if binary, otherwise False.
+     */
+    @VisibleForTesting
+    static boolean isBinaryContentType(String contentType) {
+        return contentType != null && !contentType.startsWith("text/");
+    }
+
+    /**
+     * Validates ErrorAttachmentLog
+     *
+     * @param log ErrorAttachmentLog to validate.
+     * @return True if validation succeeded, otherwise False.
```

Comment:
We don't use that convention for booleans on Javadoc, we use `<code>true</code>` and `<code>false</code>`.